### ISSUE TITLE
[SURE-6692] Fleet deploying erroring on non-declared paths

### ISSUE THREAD

**Issue by kkaempf** [rancher/fleet#1792]
### Issue description:

This seems to only affect the UI, as fleet does deploy correctly, but the rancher UI shows it as a failure?

We're trying to debug a issue a customer is having that we suspect could be related to https://github.com/rancher/fleet/issues/344. There are two similar fleet manifests, one targeting an internal cluster, and the other targets an external cluster. They both use the same git repository but with different paths, and the manifests select their respective clusters using targets clusterName.

What they're running into when deploying is the rancher interface shows deployment errors as if paths in the deployment for the external cluster are also trying to deploy to the internal cluster, and erroring out. When the client checks the deployments manually, the resources actually being created match what they configured as intended.

### Business impact:

UI falsely shows failures

### Repro steps:

The fleet definitions look like:
~~~
apiVersion: fleet.cattle.io/v1alpha1
kind: GitRepo
metadata:
  name: aaa-001-ext
  namespace: fleet-default
spec:
  branch: cat2
  clientSecretName: secretToken
  insecureSkipTLSVerify: true
  paths:
  - ./k8s-resources/config-maps/external
  - ./k8s-resources/ingresses/external
  - ./k8s-resources/network-policies/external
  - ./k8s-resources/secrets/external
  - ./session-management
  pollingInterval: 60s
  repo: https://repo_url/
  targetNamespace: customNamespaceName
  targets:
  - clusterName: cluster001
~~~
and
~~~
apiVersion: fleet.cattle.io/v1alpha1
kind: GitRepo
metadata:
  name: aaa-001
  namespace: fleet-default
spec:
  branch: cat2
  clientSecretName: secretToken
  forceSyncGeneration: 3
  insecureSkipTLSVerify: true
  paths:
  - ./k8s-resources/config-maps/internal
  - ./k8s-resources/ingresses/internal
  - ./k8s-resources/network-policies/internal
  - ./k8s-resources/persistent-volume-claims/internal
  - ./k8s-resources/secrets/internal
  - ./path1
  - ./path2
  - ./path3
  - ./path4
  pollingInterval: 60s
  repo: https://repo_url/
  targetNamespace: customNamespaceName
  targets:
  - clusterName: cluster002
~~~

### Workaround:
Is workararound available and implemented? no
What is the workaround:

### Actual behavior:
The UI shows errors, but it deploys

### Expected behavior:
Deploys without errors

**Comment by raulcabello** [comment]
Maybe related with https://github.com/rancher/fleet/issues/1812 ?  We would need to know if the Bundles created for each path have the same releaseName.

**Comment by mmartin24** [comment]
### System information:
 * Rancher version: 2.8.3-rc2
 * Fleet version: v103.0.1
 
---
### Tested
- Deployed local and downstream cluster
- Using [this](https://github.com/jordojordo/fleet-examples/tree/master/resources/fleet-timeout) example already provided in the original ticket, I deployed one cluster of external content and in other internal content with several paths.
- Observed deployment was done ok and without issues:
![2024-03-04_13-10](https://github.com/rancher/fleet/assets/37271841/86aa980e-8c2c-4dc9-a116-6e6d595c1f24)

So this issue (https://github.com/rancher/dashboard/pull/9408)  seems ok in `2.8.3-rc2` 

---
### Additional notes
- When both examples are deployed on the same cluster, it errors:
![2024-03-04_13-23](https://github.com/rancher/fleet/assets/37271841/cc858516-f66b-415c-ac73-5dd6bde35fab)

- Bundle counts are still not ok (noted in https://github.com/rancher/dashboard/issues/9866 ) and it will be addressed in 2.9 

https://github.com/rancher/fleet/assets/37271841/2d8cc737-9c3f-41fe-8c63-10675bbdb0cc
